### PR TITLE
Fix piracast to make it work with Sony Xperia M

### PR DESCRIPTION
**Pull request by manishj-patel**
Hello,

I have made changes in MPEG2 TS layer of Piracast code to make it work with Sony Xperia M. So, kindly review my patches and accept them.

Please let me know in case of any query.

Thanks,
Manish P.

### CODE REVIEW

**Comment by codemonkeyricky** [comment]
Sorry, just got back from a vacation. Looking at your merge request now.

**Comment by codemonkeyricky** [comment]
So which phones have you tested and working now? And I'm assuming the phones you've tried can all work without HDCP? Some phones I tried (Galaxy S4 I think) only works with HDCP.

**Comment by manishj-patel** [comment]
I have tested Piracast with Sony Xperia M, HTC one, Google Android phone & Sony Xperia Z1. Among of them, I know about HTC one that it supports HDCP. But as Piracast doesn't support HDCP feature,  HTC one can't share HDCP content to Piracast. Rest works well.

**Comment by manishj-patel** [comment]
Comment are added in 379c5dcc324b736f3239c5ccecab6f2d1a30a527 commit.

**Comment by manishj-patel** [comment]
Samsung Galaxy S4 can discover and pair with Piracast. But after pairing, it doesn't send any data and terminate the stream. I am not sure it is because of HDCP or because of its Allshare stack.

File: mgmt_audio_decoder/src/sx_mgmt_audio_decoder.c [hunk 1]
```diff
@@ -251,7 +253,9 @@ static UINT8 slice_start_find(
             sPES_EXT *pes_ext = (sPES_EXT *) curr_ptr;
 
             *pes_payload_size = ntohs(pes_ext->length) - 14;
-            assert(*pes_payload_size == 1920);
+
+            // PES packet length is zero for Sony Xperia
+            assert((*pes_payload_size == 1920) || (*pes_payload_size == -14));
```
Comment (bdun) [inline mgmt_audio_decoder/src/sx_mgmt_audio_decoder.c:258]: Isn't it better to do a dynamic check for Sony Xperia before the - 14 above?  As storing -14 isn't really valid as the payload size.  Could -14 being updated into the argument cause issues?

**Pull request merged**